php/blocks/controls/class-control-abstract.php

```diff
@@ -388,4 +418,27 @@ function( $option ) use ( &$options ) {
 
 		return $value;
 	}
+
+	/**
+	 * Sanitize the post type, to ensure that it's a public post type.
+	 *
+	 * This expects the rest_base of the post type, as it's easier to pass that to apiFetch in the Post control.
+	 * So this iterates through the public post types, to find if one has the rest_base equal to $value.
+	 *
+	 * @param string $value The rest_base of the post type to sanitize.
+	 * @return string|null The sanitized rest_base of the post type, or null.
+	 */
+	public function sanitize_post_type( $value ) {
+		$public_post_types = get_post_types( array( 'public' => true ) );
+		foreach ( $public_post_types as $post_type ) {
+			$post_type_object = get_post_type_object( $post_type );
+			if (
+				( $post_type_object && ( $value === $post_type_object->rest_base ) )
+				||
+				$post_type === $value
+			) {
+				return $value;
```

Comment:
This looks like it will sort-of fail to sanitise (or at least fail to return a value) if the `$value` passed in isn't a registered post type, _or_ if there are no registered post types. This could be refactored so that we return `null` or `false` or `''` or something if there's a bad result.
e.g.
```
	public function sanitize_post_type( $value ) {
		$value = '';

		$public_post_types = get_post_types( array( 'public' => true ) );
		foreach ( $public_post_types as $post_type ) {
			$post_type_object = get_post_type_object( $post_type );

			if ( $post_type_object && ( $value === $post_type_object->rest_base ) ) {
				$value = $post_type_object->rest_base;
				break;
			}

			if ( $value === $post_type ) {
				$value = $post_type;
				break;
			}
		}

		return $value;
	}
```

But maybe there's a better way?

Reply:
Hi @lukecarbis,
Good point that it doesn't return anything here. I wasn't quite sure what was best.

But it looks like having the validation method return `null` (or not return anything) is necessary to fall back to [the default value](https://github.com/getblocklab/block-lab/blob/ccabbed990b5541e5ac3ff59b1352c6d8bbe3dad/php/blocks/controls/class-control-setting.php#L114). So if the validation fails somehow, this will query for the post type of `post`.

I could see how it's strange that the function sometimes doesn't return anything. But it [looks like](https://www.php.net/manual/en/functions.returning-values.php) not returning anything means that it returns `null`:
>If the return is omitted the value NULL will be returned.

Not a strongly-held opinion, though.

Reply:
On second though, I think it is better to have a return statement at the end of the function in case it hasn't returned. I'll make a commit with that.

Reply:
https://github.com/getblocklab/block-lab/pull/257/commits/95cc6d2705959fee5f5b72b7646068657ffb146c#diff-dc4ef2232132f132dbd59b26a1f8db80R454 adds `return null` to the end of the function. I also refactored it so there's only one method that finds all of the eligible post types.